sdks/python/apache_beam/runners/portability/flink_runner_test.py

```diff
@@ -129,8 +128,8 @@ def get_runner(cls):
     def create_options(self):
       options = super(FlinkRunnerTest, self).create_options()
       options.view_as(DebugOptions).experiments = ['beam_fn_api']
-      options.view_as(FlinkOptions).parallelism = 1
-      options.view_as(FlinkOptions).shutdown_sources_on_final_watermark = True
+      options._all_options['parallelism'] = 1
+      options._all_options['shutdown_sources_on_final_watermark'] = True
```

Comment:
This becomes now a bit more intransparent. Wonder if we could keep some of the FlinkOptions but fill out the remaining by the DescribeRequest.

Reply:
Do you mean it becomes harder for a user to know what the options are? I think the source for that will be the runner documentation and we can probably also improve the help command to include the runner options.

Reply:
Non-programmatic is how users would normally specify options (using the command line). This here, being a Flink runner test is atypical and I don't think we need to optimize for it.

Reply:
Command-line options are one way, another is to use the options provided by the SDK. It would be helpful to at least know that Flink options exist, even if they are provided by the Runner at runtime. I feel like this is where the new approach falls a bit short. Just something to think about.

Reply:
How about adding an empty `RunnerOptions` class and explain where they come from there?

Reply:
Yes, this is a good idea.